### PR TITLE
[ARM/Linux] Fix HFA structs

### LINKED ISSUE #6677: [ARM/Linux] `hfa testE` Unit tests fail
All 24 of the unittests in `JIT/jit64/hfa/main/testE` fail on ARM/Linux. (does not happen on arm-softfp)

error message for `hfa_nf0E_d`:

```
FAILED   - [5076:   2s] JIT/jit64/hfa/main/testE/hfa_nf0E_d/hfa_nf0E_d.sh
               BEGIN EXECUTION
               /home/sjlee/Windows_NT.x86.Release/Tests/coreoverlay/corerun hfa_nf0E_d.exe
               [Identity HFA 01]	PASSED
               [Identity HFA 02]	FAILED => expected = 3, actual = 0
               [Identity HFA 03]	FAILED => expected = 4, actual = 0
               [Identity HFA 05]	PASSED
               [Identity HFA 08]	PASSED
               [Identity HFA 11]	PASSED
               [Identity HFA 19]	PASSED
               [Get HFA 01]	PASSED
               [Get HFA 02]	FAILED => expected = 3, actual = 0
               [Get HFA 03]	FAILED => expected = 4, actual = 0
               [Get HFA 05]	PASSED
               [Get HFA 08]	PASSED
               [Get HFA 11]	PASSED
               [Get HFA 19]	PASSED
               Expected: 100
               Actual: 1
               END EXECUTION - FAILED
```

### PR DESCRIPTION
**Pull request by wateret**
Re-enable FEATURE_MULTIREG_RET for ARM32 which was disabled by #6467.

Fix #6677

### CODE REVIEW

**Comment by lemmaa** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build

**Comment by briansull** [comment]
LGTM

File: clrdefinitions.cmake [hunk 1]
```diff
@@ -35,6 +35,7 @@ elseif (CLR_CMAKE_TARGET_ARCH_ARM)
   add_definitions(-D_TARGET_ARM_=1)
   add_definitions(-DDBG_TARGET_32BIT=1)
   add_definitions(-DDBG_TARGET_ARM=1)
+  add_definitions(-DFEATURE_MULTIREG_RETURN)
```
Comment (swaroop-sridhar) [inline clrdefinitions.cmake]: Why is defining `FEATURE_MULTIREG_RETURN` necessary? 
`FEATURE_MULTIREG_RETURN` is a VM setting, different from the JIT's `FEATURE_MULTIREG_RET`  switch.
I don't think the VM setting is relevant for HFAs

**Comment by swaroop-sridhar** [comment]
Thanks for the fix.

File: clrdefinitions.cmake (hunk 1 above)
Comment (wateret) [inline clrdefinitions.cmake]: That's the part I was not sure about. I'll get rid of this change.

**Comment by wateret** [comment]
@dotnet-bot test Windows_NT x86 ryujit Checked Build and Test

**Comment by wateret** [comment]
@briansull Ready to get merged.

**Pull request merged**